### DISCUSSION TITLE
Use multiple choice answers to define stimuli

### DISCUSSION THREAD

**Discussion by GWarlop** [jspsych/jsPsych#1357] [Q&A]
Hello! I am very new to JsPsych and I am designing an experiment in which I need to present different stimuli to left and right handed people. Therefore, I first add a multiple-choice question in the timeline to assess the participant’s handedness. Depending on their answer I need to show different stimuli and determine some other variables in my code. I made the (simplified) code for a right-handed participant and that is running smoothly, but now I am struggling to alter the stimuli depending on their answer on the multiple-choice question. Here are the relevant pieces of the code:

      /// QUESTION HANDEDNESS ///

      var handedness_question = {
        type: 'survey-multi-choice',
        questions: [
          {prompt: "Which hand is your dominant hand?", name: "handedness", options: ["Left hand", "Right hand"], required: true}
        ],
        on_finish: function(data) {
            var responses = JSON.parse(data.responses);         /// here I try to get the response, which is working, but I can’t use this variable outside of this var (I think)
            var handedness = responses["handedness"];
            console.log(handedness);
        }
      };
      timeline.push(handedness_question);

        /// DEFINING LEFT AND RIGHT RESPONSE KEYS DEPENDING ON MULTIPLE CHOICE RESPONSES ///

        if (handedness == "Left hand"){              /// so this is not working, so I tried: if(jsPsych.data.get().select(responses).value == “Left hand”), but this is also not working

          left_key = jsPsych.pluginAPI.convertKeyCharacterToKeyCode('h');
          right_key = jsPsych.pluginAPI.convertKeyCharacterToKeyCode('g');
        } else {
          left_key = jsPsych.pluginAPI.convertKeyCharacterToKeyCode('g');
          right_key = jsPsych.pluginAPI.convertKeyCharacterToKeyCode('h');
        };

      /// ARRAY THAT CONTAINS DIFFERENT TRIALS WE WANT TO RUN IN TEST PHASE

      var test_stimuli;
      if (handedness == "Left hand"){				/// I tried the same options as in the part above
        test_stimuli = [
          {img1: "Stim/LEFT HANDED/UP_CON_I_1_START.bmp", img2: "Stim/LEFT HANDED/UP_CON_I_1_END.bmp", data: {test:'exp', finger: 0, cue: 0, con: "C", correct_response: "h"}},
          {img1: "Stim/LEFT HANDED/UP_CON_M_2_START.bmp", img2: "Stim/LEFT HANDED/UP_CON_M_2_END.bmp", data: {test:'exp', finger: 1, cue: 1, con: "C", correct_response: "g"}},
          {img1: "Stim/LEFT HANDED/UP_INC_I_2_START.bmp", img2: "Stim/LEFT HANDED/UP_INC_I_2_END.bmp", data: {test:'exp', finger: 0, cue: 1, con: "I", correct_response: "g"}},
          {img1: "Stim/LEFT HANDED/UP_INC_M_1_START.bmp", img2: "Stim/LEFT HANDED/UP_INC_M_1_END.bmp", data: {test:'exp', finger: 1, cue: 0, con: "I", correct_response: "h"}}
        ];
      } else {
        test_stimuli = [
          {img1: "Stim/RIGHT HANDED/UP_CON_I_1_START.bmp", img2: "Stim/RIGHT HANDED/UP_CON_I_1_END.bmp", data: {test:'exp', finger: 0, cue: 0, con: "C", correct_response: "g"}},
          {img1: "Stim/RIGHT HANDED/UP_CON_M_2_START.bmp", img2: "Stim/RIGHT HANDED/UP_CON_M_2_END.bmp", data: {test:'exp', finger: 1, cue: 1, con: "C", correct_response: "h"}},
          {img1: "Stim/RIGHT HANDED/UP_INC_I_2_START.bmp", img2: "Stim/RIGHT HANDED/UP_INC_I_2_END.bmp", data: {test:'exp', finger: 0, cue: 1, con: "I", correct_response: "h"}},
          {img1: "Stim/RIGHT HANDED/UP_INC_M_1_START.bmp", img2: "Stim/RIGHT HANDED/UP_INC_M_1_END.bmp", data: {test:'exp', finger: 1, cue: 0, con: "I", correct_response: "g"}}
        ];
      };

I assume it has something to do with certain elements being available at certain points or not, but I can’t figure out how to access the handedness information from the multiple choice question and use it further in my code. What am I doing wrong and how can I get the handedness information from the multiple-choice question and use it further in the experiment?

**Comment by jodeleeuw** [comment]
Hey @GWarlop ,

Check out https://github.com/jspsych/jsPsych/discussions/1343 for a similar problem and possible solutions.

**Reply by GWarlop** [reply]
Hi, thank you for the help! I think I will need a dynamic parameter for timeline_variables. Like this:

      var test_procedure = {
        timeline: [if_no_response, test],
        timeline_variables: function() {
          var response = jsPsych.data.get().select('responses');
          var responses = JSON.parse(response);
          var handedness = responses["handedness"];
          if (handedness == "Left hand"){
            return test_stimuli_LEFT;
          } else {
            return test_stimuli_RIGHT;
          }
        },
        sample: {
          type: 'with-replacement',
          size: 10
        }
      };
      timeline.push(test_procedure);

But this does not seem to work: "Uncaught TypeError: Cannot read property 'img1' of undefined". 
Is that because timeline_variables won't take a function? Or am I doing something else wrong?
Thank you in advance.

**Reply by jodeleeuw** [reply]
We haven't added `function` support for `timeline_variables` yet. I think a `conditional_function` should work here if you create one timeline for the left stimuli and another for the right.
